### PR TITLE
Convert post_agent_to_master to inlineCallbacks

### PR DESCRIPTION
**Pull request by opalmer**
Similar to other recent branches the focus of this change is to convert a part of `service.py` to use inlineCallbacks.  In this case, the part which is performing a POST request to register the agent with the master.

### CODE REVIEW

**Comment by opalmer** [comment]
@guidow, PTAL.   Will start working on some tests for this soon but the general code is here.

File: pyfarm/agent/service.py [hunk 1]
```diff
@@ -564,88 +568,84 @@ def error_while_posting(failure):
...
-        # Master is up but is rejecting our request because there's something
-        # wrong with it.  Do not retry the request.
-        elif response.code >= 400:
-            svclog.error(
-                "%s accepted our POST request but responded with code %s "
-                "which is a client side error.  The message the server "
-                "responded with was %r.  Sorry, but we cannot retry this "
-                "request as it's an issue with the agent's request.",
-                self.agents_endpoint(), response.code, response.data())
+                svclog.warning("Not retrying POST to master, shutting down.")
```
Comment (guidow) [inline pyfarm/agent/service.py]: Misleading log message – the agent is not shutting down at this point.

File: pyfarm/agent/service.py [hunk 2]
```diff
@@ -77,6 +79,8 @@ class Agent(object):
     itself with the master, starting the periodic task manager,
     and handling shutdown conditions.
     """
+    reactor = reactor
```
Comment (guidow) [inline pyfarm/agent/service.py]: Why is this necessary?

**Comment by guidow** [comment]
I have looked over the code and tried the branch in practice. It works, and I could not find any real problems.

**Comment by guidow** [comment]
Works in principle, but has a few bugs left.

File: pyfarm/agent/service.py (hunk 2 above)
Comment (opalmer) [inline pyfarm/agent/service.py]: It's not anymore, removing.  Anytime you see something like this it's usually the reactor can be replaced in testing.  In the case of this change I have not had a reason to use it for testing so far...

**Comment by opalmer** [comment]
Thanks for the review.  Still finishing the tests for this so I wouldn't be surprised if there's a few issues lingering.  Are the bugs you're seeing from reviewing the code or through direct testing (trying to make sure I don't miss anything in the tests I'm writing).

**Comment by guidow** [comment]
I'm sorry, the comment with "has a few bugs left" should have gone to the other PR...

**Comment by opalmer** [comment]
Ah ok that makes sense, thanks for clarifying.  I'm going to continue adding the remaining tests for this then and once that's complete I'll go ahead and merge, assuming there are not any major changes to the code, unless you have objections.

**Comment by opalmer** [comment]
Going to go ahead and merge this since there are not any functional changes since the last review.
